draft-ietf-oauth-browser-based-apps.md

```diff
@@ -196,100 +201,138 @@ In recent years, widespread adoption of Cross-Origin Resource Sharing (CORS), wh
+
+
+
+### Acquisition and Extraction of New Tokens {#payload-new-flow}
+
+In this advanced attack scenario, the attacker completely disregards any tokens that the application has already obtained. Instead, the attacker takes advantage of the ability to run malicious code in the application's origin. With that ability, the attacker can inject a hidden iframe and launch a silent Authorization Code flow. This silent flow will reuse the user's existing session with the authorization server and result in the issuing of a new, independent set of tokens. This scenario consists of the following steps:
+- Execute malicious JS code
+- Setup a handler to obtain the authorization code from the iframe (e.g., by monitoring the frame's URL or via Web Messaging)
+- Insert a hidden iframe into the page and initialize it with an authorization request. The authorization request in the iframe will occur within the user's session and, when the session is still active, result in the issuing of an authorization code.
+- Extract the authorization code from the iframe using the previously installed handler
+- Send the authorization code to a server controlled by the attacker
+- Exchange the authorization code for a new set of tokens
+- Abuse the stolen tokens
+
+The most important takeaway from this scenario is that it runs a new OAuth flow instead of focusing on stealing existing tokens. In essence, even if the application finds a token storage mechanism with perfect security, the attacker will still be able to request a new set of tokens.
+
+This attack scenario is possible because the security of public browser-based OAuth 2.0 clients relies entirely on the redirect URI and application's origin. When the attacker can execute malicious JavaScript code in the application's execution context, they effectively control the application's origin, and by extension, the redirect URI. This attack scenario uses a silent iframe-based flow, which is the same mechanism most browser-based apps use to bootstrap their authentication state. Since the attacker controls the application in the browser, the attacker's authorization code flow is indistinguishable from a legitimate authorization code flow.
+
+There are no security mechanisms for frontend applications that counter this attack scenario. Short access token lifetimes and refresh token rotation are ineffective, since the attacker has a fresh, independent set of tokens. Advanced security mechanism, such as DPoP ({{dpop}}) are equally ineffective, since the attacker can use their own key pair to setup and use DPoP for the newly obtained tokens.
```

Comment:
```suggestion
There are no security mechanisms for frontend applications that counter this attack scenario. Short access token lifetimes and refresh token rotation are ineffective, since the attacker has a fresh, independent set of tokens. Advanced security mechanism, such as DPoP ({{DPoP}}) are equally ineffective, since the attacker can use their own key pair to setup and use DPoP for the newly obtained tokens.
```